tile/platform/local_machine/mem_cache.cc

```diff
@@ -16,11 +35,16 @@ std::shared_ptr<hal::Buffer> MemCache::TryAlloc(std::size_t size) {
     l.pop();
     return result;
   }
-  return std::shared_ptr<hal::Buffer>{};
+  Cleanup(size);
+  total_ += size;
+  auto buffer = source_->MakeBuffer(size, hal::BufferAccessMask::DEVICE_RW);
```

Comment:
Why are you moving the underlying allocation into the cache?

IMHO -- this is making the class more complicated than it needs to be; now it needs to know about memory allocation policy.